Breadcrumbs.py

```diff
@@ -1,94 +1,107 @@
 # -*- coding: utf-8 -*-
-import sublime, sublime_plugin
-import math
+import re
+import html
+
+import sublime
+import sublime_plugin
 
 try:
   xrange
 except NameError:
   xrange = range
 
+
 def get_tab_size(view):
   return int(view.settings().get('tab_size', 8))
 
-def get_row_indentation(points, view, tab_size, limit=1e20):
-  pos = 0
-  for pt in points:
-    ch = view.substr(pt)
 
+def get_row_indentation(line_start, view, tab_size, indentation_limit=None):
+  if indentation_limit is None:
+    row = view.rowcol(line_start)[0]
+    indentation_limit = view.text_point(row + 1, 0) - line_start
+
+  pos = 0
+  for ch in view.substr(sublime.Region(line_start, line_start + indentation_limit)):
     if ch == '\t':
       pos += tab_size - (pos % tab_size)
 
-    elif ch.isspace():
+    elif ch == ' ':
       pos += 1
 
     else:
       break
 
-    if limit <= pos:
+    if indentation_limit <= pos:
       break
 
   return pos
 
+
 def is_white_row(view, points):
   return all(view.substr(pt).isspace() for pt in reversed(points))
 
-def get_breadcrumb(view, points, limit):
-  for pt in points:
-    ch = view.substr(pt)
-    if not ch.isspace():
-      return view.substr(sublime.Region(pt, min(view.line(pt).b, pt + limit))).strip()
-  return ''
 
-class BreadcrumbsCommand(sublime_plugin.EventListener):
-  def on_selection_modified(self, view):
-    tab_size = get_tab_size(view)
-    settings = sublime.load_settings('Breadcrumbs.sublime-settings')
-    breadcrumb_length_limit = settings.get('breadcrumb_length_limit', 100)
-    separator = settings.get('breadcrumbs_separator', u' ■ ')
-    total_breadcrumbs_length_limit = settings.get('total_breadcrumbs_length_limit', 200)
-    view.erase_status('breadcrumbs')
+def get_breadcrumb(view, points, regex, limit):
+  linestring = view.substr(sublime.Region(min(points), min(view.line(min(points)).b, min(points) + limit)))
+  match = re.search(regex, linestring)
+  if match:
+    return(match.group('name'))
+  else:
+    return None
 
-    if len(view.sel()) == 0:
-      return
 
-    current_row = view.rowcol(view.sel()[0].b)[0]
+def make_breadcrumbs(view, current_row, shorten=False):
+  if len(view.sel()) == 0:
+    return
 
-    def get_row_start(row):
-      return view.text_point(row,0)
+  settings = sublime.load_settings('Breadcrumbs.sublime-settings')
+  tab_size = get_tab_size(view)
+  default_breadcrumb_regex = settings.get('breadcrumb_regex', u'(?P<name>.*)')
+  breadcrumb_regex = view.settings().get('breadcrumb_regex', default_breadcrumb_regex)
+  separator = settings.get('breadcrumbs_separator', u' ')
```

Comment:
I'm not sure what is the convention for defaults. My original idea was to keep three things in sync:

- the default value passed to `settings.get` as the second argument
- the default value in the Breadcrumbs.sublime-settings file
- the default value described in the table in README.md

I know this is not DRY. But it somehow better than allowing any two of this places to be out of sync, which would in my opinion be confusing.

Thinking about it, perhaps a better way to handle it would be:
- remove the table from README.md, and replace it with a sentence like "See the well commented Breadcrumbs.sublime-settings to learn about multiple possible ways to customize it!"
- add comments to the Breadcrumbs.sublime-settings explaining each option in detail, with examples etc.
- throw an exception from the plugin if any of the config values is not provided at all

This way we would have only one place to define default values. But is it a standard way of doing things?